src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -428,6 +438,70 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
             }
         }
 
+        // gets metadata from worker
+        public Task<List<FunctionMetadata>> WorkerGetFunctionMetadata()
+        {
+            return SendWorkerMetadataRequest();
+        }
+
+        internal Task<List<FunctionMetadata>> SendWorkerMetadataRequest()
+        {
+            _eventSubscriptions.Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.WorkerFunctionMetadataResponse)
+                        .Timeout(_functionLoadTimeout)
+                        .Take(1)
+                        .Subscribe((msg) => ProcessMetadata(msg.Message.WorkerFunctionMetadataResponse), HandleWorkerMetadataRequestError));
+
+            _workerChannelLogger.LogDebug("Sending WorkerMetadataRequest to {language} worker with worker ID {workerID}", _runtime, _workerId);
+
+            // sends the function app directory path to worker for indexing
+            SendStreamingMessage(new StreamingMessage
+            {
+                WorkerFunctionMetadataRequest = new WorkerFunctionMetadataRequest()
+                {
+                    Directory = _applicationHostOptions.CurrentValue.ScriptPath
+                }
+            });
+            return _functionsIndexingTask.Task;
+        }
+
+        // parse metadata response into FunctionMetadata objects
+        internal void ProcessMetadata(WorkerFunctionMetadataResponse workerFunctionMetadataResponse)
+        {
+            _workerChannelLogger.LogDebug("Received the worker function metadata response from worker {worker_id}", _workerId);
+
+            var responseStatus = workerFunctionMetadataResponse.OverallStatus;
+
+            var functions = new List<FunctionMetadata>();
+
+            foreach (var metadata in workerFunctionMetadataResponse.Results)
+            {
+                if (metadata.Status != null && metadata.Status.IsFailure(out Exception metadataRequestEx))
+                {
+                    _workerChannelLogger.LogDebug($"Worker failed to index function {metadata.Id}");
```

Comment:
Change this to `LogError()` -- then should be all good!

Reply:
Good call, just made the change!